src/controllers/expense.controller.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const {
+  getAll,
+  getById,
+  create,
+  remove,
+  update,
+} = require('../services/expense.service');
+const { getAll: getAllUsers } = require('../services/user.service');
+
+const parseQueryParameters = (req) => {
+  const userId = req.query.userId ? Number(req.query.userId) : null;
+  const from = req.query.from ? new Date(req.query.from) : null;
+  const to = req.query.to ? new Date(req.query.to) : null;
+  const categories = req.query.categories
+    ? req.query.categories.split(',')
+    : [];
+
+  return {
+    userId,
+    from,
+    to,
+    categories,
+  };
+};
+
+const filterExpenses = (expenses, userId, from, to, categories) => {
+  let filteredExpenses = expenses;
+
+  if (userId !== null) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === userId,
+    );
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses.filter((expense) => {
+      const spentAt = new Date(expense.spentAt);
+
+      return spentAt >= from && spentAt <= to;
+    });
+  }
+
+  if (categories.length > 0) {
+    filteredExpenses = filteredExpenses.reduce((acc, expense) => {
+      if (categories.includes(expense.category)) {
+        acc.push(expense);
+      }
+
+      return acc;
+    }, []);
+  }
+
+  return filteredExpenses;
```

Comment:
Controllers are just the entry point for requests and are responsible for sending back the response to the client. Services, on the other hand, contain the business logic. So move business logic to services